src/Controls/src/Core/HandlerImpl/Application/Application.Impl.cs

```diff
@@ -111,10 +111,8 @@ public virtual void CloseWindow(Window window)
 
 		void IApplication.ThemeChanged()
 		{
-			if (UserAppTheme != AppTheme.Unspecified)
-				return;
-
-			TriggerThemeChangedActual();
+			PlatformAppTheme = AppInfo.RequestedTheme;
+			PlatformLayoutDirection = AppInfo.RequestedLayoutDirection;
```

Comment:
This is called from Core when the OS detects some theme/settings/config change. I am potentially abusing the method for RTL, but they all seem to go here anyway.